java/vector/src/main/java/org/apache/arrow/vector/complex/RunEndEncodedVector.java

```diff
@@ -0,0 +1,683 @@
+   * @return a new {@link org.apache.arrow.vector.util.TransferPair transfer pair} that is used to
+   *     transfer underlying buffers into the target vector.
+   */
+  @Override
+  public TransferPair makeTransferPair(ValueVector target) {
+    throw new UnsupportedOperationException(
+        "RunEndEncodedVector does not support makeTransferPair(ValueVector)");
+  }
+
+  /**
+   * Get a reader for this vector.
+   *
+   * @return a {@link org.apache.arrow.vector.complex.reader.FieldReader field reader} that supports
+   *     reading values from this vector.
+   */
+  @Override
+  public FieldReader getReader() {
+    throw new UnsupportedOperationException("Not yet implemented.");
+  }
+
+  /**
+   * Get a writer for this vector.
+   *
+   * @return a {@link org.apache.arrow.vector.complex.writer.FieldWriter field writer} that supports
+   *     writing values to this vector.
+   */
+  public FieldWriter getWriter() {
+    throw new UnsupportedOperationException("Not yet implemented.");
+  }
+
+  /**
+   * Get the number of bytes used by this vector.
+   *
+   * @return the number of bytes that is used by this vector instance.
+   */
+  @Override
+  public int getBufferSize() {
+    int bufferSize = 0;
+    for (FieldVector v : getChildrenFromFields()) {
+      bufferSize += v.getBufferSize();
+    }
+    return bufferSize;
+  }
+
+  /**
+   * Returns the number of bytes that is used by this vector if it holds the given number of values.
+   * The result will be the same as if setValueCount() were called, followed by calling
+   * getBufferSize(), but without any of the closing side-effects that setValueCount() implies wrt
+   * finishing off the population of a vector. Some operations might wish to use this to determine
+   * how much memory has been used by a vector so far, even though it is not finished being
+   * populated.
+   *
+   * @param valueCount the number of values to assume this vector contains
+   * @return the buffer size if this vector is holding valueCount values
+   */
+  @Override
+  public int getBufferSizeFor(int valueCount) {
+    return 0;
+  }
+
+  /**
+   * Return the underlying buffers associated with this vector. Note that this doesn't impact the
+   * reference counts for this buffer so it only should be used for in-context access. Also note
+   * that this buffer changes regularly thus external classes shouldn't hold a reference to it
+   * (unless they change it).
+   *
+   * @param clear Whether to clear vector before returning; the buffers will still be refcounted;
+   *     but the returned array will be the only reference to them
+   * @return The underlying {@link ArrowBuf buffers} that is used by this vector instance.
+   */
+  @Override
+  public ArrowBuf[] getBuffers(boolean clear) {
+    return null;
+  }
+
+  /**
+   * Gets the underlying buffer associated with validity vector.
+   *
+   * @return buffer
+   */
+  @Override
+  public ArrowBuf getValidityBuffer() {
+    return null;
```

Comment:
Just as a note, for StringView, we used unsupported. Maybe we could keep things consistent? Or file an issue for refactor these into a common pattern?

Reply:
@vibhatha what do you mean "for StringView, we used unsupported", I can see getValidityBuffer is implemented in BaseVariableWidthViewVector.

Reply:
Ah I think I wasn't clear. Sorry about that. For instance, we don't have a `offsetBuffer` for StringView. And there rather than returning null, we have added `throw new UnsupportedOperationException(...)`. I was referring to that.

Reply:
Oh, i see.

Reply:
Oops. Yes, please make all of these (except for getBuffers) throw instead of returning null. (And I think `getBuffers` should maybe return an empty array instead of null.)

Reply:
fixed in [14ba5bb](https://github.com/apache/arrow/pull/43888/commits/14ba5bbff29398ab8772ac3ae2817ee744e270b4)